cmd/snap-update-ns/change.go

```diff
@@ -330,10 +330,20 @@ func (c *Change) lowLevelPerform(as *Assumptions) error {
 			flags := umountNoFollow
 			if c.Entry.XSnapdDetach() {
 				flags |= syscall.MNT_DETACH
+				// If we are detaching something then before performing the actual detach
```

Comment:
I think I'd make this comment more terse:
```
// Do not propagate unmount events. Prevents detach from conflicting with
// itself and getting EBUSY when hierarchies are nested. See umount(2) for details.
```

Reply:
I can adjust that if you want :-)
I *may* do so in the follow-up, if that is green and it gets approved.